### PR TITLE
Handle unfetchable Go versions that have name collisions.

### PR DESCRIPTION
**Pull request by tiegz**
Ignore versions that have case-insensitive name collisions instead of blowing up when we try to fetch their info.

* [example Go Module version with name collision](https://github.com/ysweid/aws-sdk-go/tree/v1.12.68/models/apis)
* [breaking url for ^](https://proxy.golang.org/github.com/ysweid/aws-sdk-go/@v/v1.12.68.info)
* [Go issue](https://github.com/golang/go/issues/33778)

### CODE REVIEW

File: app/models/package_manager/go.rb [hunk 1]
```diff
@@ -106,6 +106,10 @@ def self.versions(project, _name)
             known_versions[v].slice(:number, :published_at)
           else
             info = get("#{PROXY_BASE_URL}/#{project[:name]}/@v/#{v}.info")
```
Comment (jsonperl) [inline app/models/package_manager/go.rb:108]: Any way you can think of to correct to the proper case before updating? Some sort of case insensitive search mechanism?

Comment (tiegz) [reply app/models/package_manager/go.rb:108]: It's unfortunately an issue with the files in the Go Module and not with the name itself so I think we have to wait on a fix from Golang 😬 Here's the response for the example one above, where it has two files in the same folder with same-name-different-casing:

`not found: create zip: case-insensitive file name collision: "models/apis/GuardDuty/2017-11-28/api-2.json" and "models/apis/guardduty/2017-11-28/api-2.json"`

Comment (tiegz) [reply app/models/package_manager/go.rb:108]: (or alternatively we could keep the version but just display its published_at as `?` or something?)

Comment (jsonperl) [reply app/models/package_manager/go.rb:108]: Seems like a really really small amount of packages huh? (Like 2 maybe?)

**Review by jsonperl — APPROVED** [review]

File: app/models/package_manager/go.rb (hunk 1 above)
Comment (tiegz) [reply app/models/package_manager/go.rb:108]: I think this example had 3 versions with name collisions, but we've got a ton of these errors in bugsnag: https://app.bugsnag.com/tidelift/libraries-dot-io/errors/60076c589e7ce30018c6b586

**Pull request merged**